src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -963,7 +963,16 @@ void DynamoView_KeyDown(object sender, KeyEventArgs e)
 
         void DynamoView_KeyUp(object sender, KeyEventArgs e)
         {
-            if (e.Key == Key.Escape)
+            if (e.Key == Key.Escape && dynamoViewModel.WatchEscapeIsDown)
+            {
+                dynamoViewModel.WatchEscapeIsDown = false;
+                dynamoViewModel.EscapeCommand.Execute(null);
+            }
+        }
+
+        void DynamoView_LostFocus(object sender, EventArgs e)
```

Comment:
I'm just curious, does this get call if user clicks on the `Search` text box on the library UI? How about clicking in an existing `Code Block` or `Number` node?